CMakeLists.txt

```diff
@@ -84,8 +84,10 @@ target_link_directories(onnxruntime-genai PRIVATE ${ORT_LIB_DIR})
 
 # we keep the shared libraries disconnected on Android as they will come from separate AARs and we don't want to force
 # the ORT version to match in both.
-if(NOT (CMAKE_SYSTEM_NAME STREQUAL "Android" OR CMAKE_SYSTEM_NAME STREQUAL "Linux"))
-target_link_libraries(onnxruntime-genai PRIVATE ${ONNXRUNTIME_LIB})
+if(CMAKE_SYSTEM_NAME STREQUAL "Android" OR CMAKE_SYSTEM_NAME STREQUAL "Linux" OR (CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND (NOT BUILD_APPLE_FRAMEWORK)))
```

Comment:
Enable `dlopen` on macOS and **NOT** building for apple framework. `BUILD_APPLE_FRAMEWORK` will be added in future PRs.

On iOS the `CMAKE_SYSTEM_NAME` would be `iOS`.